src/components/side-panel/layer-panel/dataset-layer-section.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import SourceDataCatalogFactory from '../common/source-data-catalog';
+import LayerListFactory from './layer-list';
+
+const DatasetLayerSectionWrapper = styled.div`
+  margin-bottom: 16px;
+`;
+
+DatasetLayerSectionFactory.deps = [SourceDataCatalogFactory, LayerListFactory];
+
+function DatasetLayerSectionFactory(SourceDataCatalog, LayerList) {
+  const DatasetLayerSection = props => {
+    const {
+      dataset,
+      showDatasetTable,
+      layers,
+      updateTableColor,
+      showDeleteDataset,
+      removeDataset,
+      layerOrder,
+      layerClasses,
+      uiStateActions,
+      visStateActions
+    } = props;
+
+    const datasets = {[dataset.id]: dataset};
```

Comment:
`useMemo`

Reply:
Are there some extreme cases of loaded datasets?
I mean, won't the useMemo overhead just slow things down over this simple operation?
Just my guess, list of datasets won't be longer than let's say 20 (I'm not sure what is the extreme case).
Using useMemo would probably have to do a lot of checks on which we lose time on each render, won't we?

Reply:
Also, this here is just a single dataset being used to construct a new object.

Reply:
The part that it constructs a new object is the problem here. because `datasets` is always a new object, `SourceDataCatalog` and `LayerList` will be re-rendered all the time. If the component is small, or a leaf component, it might not be too bad, But here. it's a quite high lever component. It renders all the layer panel via the `LayerList` component. which means, all the layer panel component will be rerendered. because of the construct of a new object inside render function

Reply:
Yeah, true, totally forgot about the children.